### PR TITLE
feat: add audio support to the new previewer

### PR DESCRIPTION
**Pull request by BrayanDSO**
## Fixes
* Part of #14378 

## Approach
In the commits

## How Has This Been Tested?

[Screen_recording_20240207_185720.webm](https://github.com/ankidroid/Anki-Android/assets/69634269/0cbca9ce-a642-4455-b13c-0ef4e2d78457)

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by BrayanDSO** [comment]
Pending tasks for completion currently are:
- enable fullscreen for videos (will need some custom handling of configuration changes)
- re-implement #15217 here (not sure what is the appropriate way of doing that in a Fragment)

File: AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/SoundTagPlayer.kt [hunk 1]
```diff
@@ -74,7 +74,7 @@ class SoundTagPlayer(private val soundUriBase: String) {
                     continuation.resume(Unit)
                 }
             }
-            val soundUri = Uri.parse(soundUriBase + tag.filename)
+            val soundUri = Uri.parse(soundUriBase + Uri.encode(tag.filename))
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/SoundTagPlayer.kt:77]: This looked much more reasonable, thanks!

**Review by david-allison — APPROVED** [review]
All looks good, not even a nit, thanks!

**Review by lukstbit — APPROVED** [review]
LGTM

**Pull request merged**